pom.xml

```diff
@@ -136,6 +136,21 @@
                 <groupId>org.jenkins-ci.plugins.workflow</groupId>
                 <artifactId>workflow-support</artifactId>
                 <version>865.v43e78cc44e0d</version>
+            </dependency>
```

Comment:
I recommend removing all `<dependency>` parts of the XML here if the `groupId` starts with `org.jenkins-ci.plugins`. Most, if not all, of these have their versions set in the Jenkins Bill of Materials (BOM). This ensures we are always current with what the upstream community considers validated/tested.

Reply:
Jenkins BOM is set here: https://github.com/jenkinsci/bom

Reply:
@adambkaplan  Hi, I removed all `<dependency>` which groupId starts with `org.jenkins-ci.plugins` however, I am getting below compilation error while building it. Please suggest, if i am missing anything or how we can handle this.
I also tried with old version of Jenkins BOM and updating dependancies with sequence but still the error message getting reported. 

Error:

```

[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] <local_path>/jenkins-sync-plugin/src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java:[488,16] unreported exception hudson.model.Descriptor.FormException; must be caught or declared to be thrown
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.482 s
[INFO] Finished at: 2024-10-15T21:30:59+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project openshift-sync: Compilation failure
[ERROR] <local_path>/jenkins-sync-plugin/src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java:[488,16] unreported exception hudson.model.Descriptor.FormException; must be caught or declared to be thrown

```
Diff :

```
diff --git a/pom.xml b/pom.xml
index 69b2057..59d116b 100644
--- a/pom.xml
+++ b/pom.xml
@@ -118,7 +118,7 @@
             <dependency>
                 <groupId>io.jenkins.tools.bom</groupId>
                 <artifactId>bom-2.440.x</artifactId>
-                <version>3387.v0f2773fa_3200</version>
+                <version>3435.v238d66a_043fb_</version>
                 <scope>import</scope>
                 <type>pom</type>
             </dependency>
@@ -127,46 +127,6 @@
                 <artifactId>joda-time</artifactId>
                 <version>2.12.5</version>
             </dependency>
-            <dependency>
-                <groupId>org.jenkins-ci.plugins.workflow</groupId>
-                <artifactId>workflow-multibranch</artifactId>
-                <version>773.vc4fe1378f1d5</version>
-            </dependency>
-            <dependency>
-                <groupId>org.jenkins-ci.plugins.workflow</groupId>
-                <artifactId>workflow-support</artifactId>
-                <version>865.v43e78cc44e0d</version>
-            </dependency>
....
.......
+            </dependency>
+             <dependency>
+                <groupId>io.jenkins.plugins</groupId>
+                <artifactId>asm-api</artifactId>
+                <version>9.6-3.v2e1fa_b_338cd7</version>
+            </dependency>
+            <dependency>
+                <groupId>io.jenkins.plugins.mina-sshd-api</groupId>
+                <artifactId>mina-sshd-api-common</artifactId>
+                <version>2.12.1-113.v4d3ea_5eb_7f72</version>
             </dependency>
             <dependency>
-                <groupId>org.jenkins-ci.plugins</groupId>
-                <artifactId>scm-api</artifactId>
-                <version>676.v886669a_199a_a_</version>
+                <groupId>io.jenkins.plugins.mina-sshd-api</groupId>
+                <artifactId>mina-sshd-api-core</artifactId>
+                <version>2.12.1-113.v4d3ea_5eb_7f72</version>
             </dependency>
```

Reply:
> [ERROR] <local_path>/jenkins-sync-plugin/src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java:[488,16] unreported exception hudson.model.Descriptor.FormException; must be caught or declared to be thrown

The likely cause is that when we updated our dependencies, a function called in `CredentialsUtils.java` changed its behavior to throw a `hudson.model.Descriptor.FormException`. This is a pretty common thing to encounter when updating dependencies in Java, and similar types of challenges happen in every programming language/ecosystem.

You now need to change `CredentialsUtils.java` to handle the exception. A modern IDE like IntelliJ or Visual Studio Code (with Microsoft or Red Hat's Java extension) will likely flag this as a compile error and provide tips on how to address it.